core/src/main/java/org/apache/accumulo/core/fate/zookeeper/ZooCache.java

```diff
@@ -400,7 +458,7 @@ public byte[] run() throws KeeperException, InterruptedException {
            * non-existence can not be cached.
            */
           try {
-            Stat stat = zk.exists(zPath, watcher);
+            Stat stat = zk.exists(zPath, null);
```

Comment:
The reason these zk.exists calls were made was so that non-existence of a node could be cached. When asking for children or data for a node that did not exist no watcher would ever be set, so needed call exists to set watches on non existent nodes. With persistent recursive it seems like the need to call exists goes away and this call could be removed. That seems like it would cut the RPCs that ZooCache makes to ZooKeeper servers in half which is kinda nice.

Reply:
There is another call to exists elsewhere that could also be removed.  We have test for caching non-existence also I believe so those can help get some basic insight on remove these calls.